src/main/java/io/odpf/firehose/sink/blob/BlobSinkFactory.java

```diff
@@ -65,7 +65,7 @@ private static LocalStorage getLocalFileWriterWrapper(BlobSinkConfig sinkConfig,
     }
 
     public static BlobStorage createSinkObjectStorage(BlobSinkConfig sinkConfig, Map<String, String> configuration) {
-        if (sinkConfig.getBlobStorageType() == BlobStorageType.GCS) {
+        if (sinkConfig.getBlobStorageType() == BlobStorageType.GCS || sinkConfig.getBlobStorageType() == BlobStorageType.S3) {
```

Comment:
this should be different configuration. GCS_TYPE is for Blob sink OR DLQ type of GCS.